src/Compilers/CSharp/Test/CommandLine/SarifV2ErrorLoggerTests.cs

```diff
@@ -651,5 +651,628 @@ class C
+                              "properties": {
+                                "warningLevel": 3
+                              }
+                            }
+                          ],
+                          "tool": {
+                            "driver": {
+                              "name": "Microsoft (R) Visual C# Compiler",
+                              "version": "4.9.0-dev (<developer build>)",
+                              "dottedQuadFileVersion": "42.42.42.42",
+                              "semanticVersion": "42.42.42",
+                              "language": "en",
+                              "rules": [
+                                {
+                                  "id": "CS5001",
+                                  "defaultConfiguration": {
+                                    "level": "error"
+                                  },
+                                  "helpUri": "https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS5001)",
+                                  "properties": {
+                                    "category": "Compiler",
+                                    "tags": [
+                                      "Compiler",
+                                      "Telemetry",
+                                      "NotConfigurable"
+                                    ]
+                                  }
+                                },
+                                {
+                                  "id": "CS0169",
+                                  "shortDescription": {
+                                    "text": "Field is never used"
+                                  },
+                                  "helpUri": "https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS0169)",
+                                  "properties": {
+                                    "category": "Compiler",
+                                    "tags": [
+                                      "Compiler",
+                                      "Telemetry"
+                                    ]
+                                  }
+                                }
+                              ]
+                            }
+                          },
+                          "columnKind": "utf16CodeUnits"
+                        }
+                      ]
+                    }
+                    """,
+                    actualOutput,
+                    message: actualOutput);
+            }
+
+            CleanupAllGeneratedFiles(sourceFile);
+            CleanupAllGeneratedFiles(errorLogFile);
+        }
+
+        [ConditionalTheory(typeof(WindowsOnly), Reason = "https://github.com/dotnet/roslyn/issues/30289")]
+        [CombinatorialData]
+        public void LineDirective(SarifTestVersion version)
+        {
+            var errorLogDir = Temp.CreateDirectory();
+
+            var source = $$"""
+public class C
+{
+#line 123 "D:\otherfile.cs"
```

Comment:
This makes me a bit nervous cause there are cases where `#line` can behave differently if it can / can't find a file. Hard coding a path like this means it's possible on some machines the file may exist while on others it won't. Prefer instead if we definitively do / don't put a file on disk at this location via `Temp.CreateDirectory().CreateFile` and then use interpolation to put the known file path.